src/models/ObjectLevelCan.ts

```diff
@@ -1,17 +1,21 @@
-type BasicPermissions = 'create' | 'read' | 'update' | 'delete'
-
-
 type ObjectLevelCan<TPermissions extends string> = {
   [key in TPermissions]: boolean
 }
 
-export type BasicPermissionsObjectLevelCan = ObjectLevelCan<BasicPermissions>
-export type BasicRunnablePermissionsObjectLevelCan = ObjectLevelCan<BasicPermissions | 'run'>
+export const BasicObjectLevelPermissions = ['create', 'read', 'update', 'delete'] as const
+export const RunnableObjectLevelPermissions = [...BasicObjectLevelPermissions, 'run'] as const
+
+export type BasicPermissionsObjectLevelCan = ObjectLevelCan<typeof BasicObjectLevelPermissions[number]>
+export type BasicRunnablePermissionsObjectLevelCan = ObjectLevelCan<typeof RunnableObjectLevelPermissions[number]>
 
-export function createObjectLevelCan<T extends Record<string, boolean>>(): T {
+export function createObjectLevelCan<T extends Record<string, boolean>>(knownProperties: typeof BasicObjectLevelPermissions | typeof RunnableObjectLevelPermissions): T {
```

Comment:
this could be simpler if I just default to the superset of expected permissions (which would atm be `RunnableObjectLevelPermissions`) but this matches `createCan` more closely and leaning toward explicit over implicit here.

Reply:
I also commented on this. I don't feel strongly, but IMO the extra complexity to distinguish between runnable and non runnable objects isn't worth it here. The types are enough protection for me. But I don't feel strongly.